Updated build system

tnagorra opened the pull request:
Depends on https://github.com/the-deep/deeper/pull/965
Depends on https://github.com/the-deep/server/pull/890

## Get started

### Update on deeper
- Pull latest changes
- Add these to .env on deeper
```bash
GRAPHQL_CODEGEN_ENDPOINT=http://localhost:8001/graphql
REACT_APP_GRAPHQL_ENDPOINT=http://localhost:8001/graphql
```
- Please update port 8000 to 8001 for all other configurations in .env
```bash
# Add / replace these keys
REACT_APP_API_END=localhost:8001
REACT_APP_ADMIN_END=localhost:8001
REACT_APP_API_HTTPS=http
REACT_APP_DEEP_ENVIRONMENT=development-nav
```

### Run proxy server to connect with other development server
- Clone https://github.com/toggle-corp/simple-proxy
- Add these to .env on simple-proxy
```bash
SOURCE_DOMAIN=nav-local.togglecorp.com
SOURCE_PROTOCOL=https
```
- Run `docker-compose up`

### Using client
- Build docker using
```bash
docker-compose build
```
- We don't have a login page yet
- Login using admin panel
- Oh no! you don't have username and password. Please contact @thenav56 jee!
- Run using `docker-compose up client`
- Don't forget to generate introspection schema

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

tnagorra committed:
Remove un-necessary files

tnagorra committed:
Add config and base setup for new build

thenav56 commented on .github/workflows/ci.yml:
```diff
@@ -11,14 +11,6 @@ jobs:
     name: 🚴 Build + Lint + Test 🚴  # Match the name below (8398a7/action-slack).
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
```
This is needed to fetch the-deep/client

tnagorra replied:
Thanks!

tnagorra committed:
Fix github workflow

- Made sure yarn generate works everywhere

tnagorra committed:
Add run_develop.sh

AdityaKhatri commented on package.json:
```diff
@@ -1,216 +1,127 @@
 {
-    "name": "deep",
+    "name": "base-react-app",
```
Rename to DEEP

AdityaKhatri commented on package.json:
```diff
@@ -1,216 +1,127 @@
-        "@the-deep/deep-ui/build/index.css",
-        "./src/Multiplexer/styles.scss"
-    ],
+    "description": "Base React Application",
```
Rename to DEEP

AdityaKhatri commented on package.json:
```diff
@@ -1,216 +1,127 @@
+    ],
+    "repository": {
+        "type": "git",
+        "url": "git+https://github.com/togglecorp/base-react-app.git"
```
Rename to DEEP

AdityaKhatri commented on package.json:
```diff
@@ -1,216 +1,127 @@
+    "author": "",
+    "license": "ISC",
+    "bugs": {
+        "url": "https://github.com/togglecorp/base-react-app/issues"
```
Rename to DEEP

tnagorra replied:
okay sir!

AdityaKhatri reviewed (changes requested):
Rename base-react-app to DEEP and properly maintain brand name.
Also, please rebase with develop-deep and send a PR.

AdityaKhatri commented on .gitignore:
```diff
@@ -2,8 +2,6 @@
 
 # dependencies
 /node_modules
-vendor/
```
Add this back for now

tnagorra committed:
Add support for project level routes

- Move ready and errored states to Init
- Init can now define context providers to app
- Separate out navbar context
- Create project context
- Update sentry configuration for development
- Create nested navigation for project related routes
- Add support for absolute path for nested routes
- Update checkPermissions to include Project
- Get current project information and permissions

frozenhelium committed:
Improve simplified text view

frozenhelium committed:
Add navbar context

tnagorra committed:
Get lastActiveProject information

- Pass projectId to useRouteMatching

tnagorra committed:
Update name on package.json

frozenhelium committed:
Add TagginFlow and Sources views and their dependencies

frozenhelium committed:
Remove old request

tnagorra committed:
Add provider for request

- Fix import errors for request

frozenhelium committed:
Add dependencies and route for tagging flow

tnagorra committed:
Pass CSRF in request

frozenhelium committed:
Make tagging flow work in new UI

frozenhelium committed:
Improve styling for sources view

tnagorra committed:
Fix typing issues

frozenhelium committed:
Setup page for sources grid view

tnagorra committed:
Replace reverseRoute with generatePath

frozenhelium committed:
Fix linting issuess

frozenhelium committed:
Add typings for browser extension
